.github/workflows/lint.yaml

```diff
@@ -0,0 +1,44 @@
+name: Lint
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+
+jobs:
+  rustfmt:
+    name: Check Rust formatting
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Install Rust
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
+          profile: minimal
+          override: true
+          components: rustfmt
+
+      - run: cargo fmt -- --check
+
+  clippy:
+    name: Check Clippy
+    runs-on: windows-latest
```

Comment:
Could you add a comment here saying why Windows is needed? I feel like leaving it as a commit title would get lost over time.